app/Album.php

```diff
@@ -246,9 +245,94 @@ public function update_min_max_takestamp()
 	}
 
 	/**
-	 * Apply the previous method on each album in the database
-	 * TODO: this seems expensive and avoidable (should only need to update
-	 * the albums that were modified, plus recursively their parents).
+	 * Update album's min_takestamp and max_takestamp based on changes made
+	 * to the album content.  If needed, recursively updates parent album(s).
+	 *
+	 * @param array $takestamps: an array with the takestamps of changed
+	 *                           elements; for albums needs to include both min and max takestamps
+	 *                           (including null elements in the array is safe)
+	 * @param bool  $adding:     true if adding new content, false if removing
+	 */
+	public function update_takestamps(array $takestamps, bool $adding)
```

Comment:
This function should return true of false depending on failure.

Reply:
Should be fixed with the latest commit...